ray-operator/controllers/ray/utils/util.go

```diff
@@ -71,6 +72,35 @@ func IsCreated(pod *corev1.Pod) bool {
 	return pod.Status.Phase != ""
 }
 
+func FindPodReadyCondition(pod *corev1.Pod, condType rayv1.RayClusterConditionType) metav1.Condition {
+	replicaPodReadyCondition := metav1.Condition{
```

Comment:
Is it possible for `pod.Status.Conditions` not to have `corev1.PodReady`? If so, we will create a condition with an empty `Reason`.

Reply:
yes you are right, i update the logic here to have an initial Condition
https://github.com/ray-project/kuberay/blob/01e90f67a4d235461f2847907bfa5b096694dee7/ray-operator/controllers/ray/utils/util.go#L75-L79

plus this logic 

https://github.com/ray-project/kuberay/blob/01e90f67a4d235461f2847907bfa5b096694dee7/ray-operator/controllers/ray/utils/util.go#L95-L98

to prevent empty reason sneak into later `SetStatusCondition`